### PR TITLE
[sqlserver] Support for SQLServer Dialect

### PR DESCRIPTION
**Pull request by fakirAyoub**
* redefinition of regex expressions
* add a `column name` parameter to some function in the dialect

### CODE REVIEW

File: packages/sqlserver/sodasql/dialects/sqlserver_dialect.py [hunk 1]
```diff
@@ -86,13 +86,13 @@ def sql_connection_test(self):
...
         try:
             conn = pyodbc.connect(
-                ('Trusted_Connection=YES;' if self.trusted_connection else '') +
-                'DRIVER={' + self.driver +
-                '};SERVER=' + self.host +
-                ';PORT=' + self.port +
-                ';DATABASE=' + self.database +
-                ';UID=' + self.username +
-                ';PWD=' + self.password)
+                ('Trusted_Connection=NO;' if True else '') +
```
Comment (vijaykiran) [inline packages/sqlserver/sodasql/dialects/sqlserver_dialect.py:89]: Please revert these .. trusted connection is required to make sure that AD SSO works.

File: packages/sqlserver/sodasql/dialects/sqlserver_dialect.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def default_connection_properties(self, params: dict):
             'port': '1433',
             'username': 'env_var(SQLSERVER_USERNAME)',
             'password': 'env_var(SQLSERVER_PASSWORD)',
-            'database': params.get('database', 'your_database'),
+            'database': params.get('database', 'sodasqldb'),
```
Comment (vijaykiran) [inline packages/sqlserver/sodasql/dialects/sqlserver_dialect.py:65]: I guess you didn't intend to leave it here?

**Review by vijaykiran — CHANGES_REQUESTED** [review]
I'm not sure if we want to skip the tests for all other warehouses, because SQL Server isn't working .. I need to think about a better strategy. May be parametrise the tests and skip only for SQLServer with a message?

File: packages/sqlserver/sodasql/dialects/sqlserver_dialect.py (hunk 1 above)
Comment (fakirAyoub) [reply packages/sqlserver/sodasql/dialects/sqlserver_dialect.py:89]: Yes I will be reversing these, hence the WIP

File: packages/sqlserver/sodasql/dialects/sqlserver_dialect.py (hunk 2 above)
Comment (fakirAyoub) [reply packages/sqlserver/sodasql/dialects/sqlserver_dialect.py:65]: this is just my local db that I used for testing

File: tests/local/warehouse/validity/test_date_and_time_validity_formats.py [hunk 3]
```diff
@@ -21,7 +22,7 @@ class TestDateAndTimeValidityFormats(SqlTestCase):
     def test_date_eu(self):
         self.sql_recreate_table(
             [f"name {self.dialect.data_type_varchar_255}"],
-            ["('21-01-2021')",
```
Comment (vijaykiran) [inline tests/local/warehouse/validity/test_date_and_time_validity_formats.py:24]: Any reason why this is removed? I think we support date validation with `-` in it

**Pull request merged**